src/MarkdownWriter/ModulePageWriter.cs

```diff
@@ -0,0 +1,146 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.PowerShell.PlatyPS.Model;
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.IO;
+using System.Text;
+
+namespace Microsoft.PowerShell.PlatyPS.MarkdownWriter
+{
+    internal class ModulePageWriter
+    {
+        private string _modulePagePath;
+        private StringBuilder sb;
+        private readonly Encoding _encoding;
+
+        public ModulePageWriter(MarkdownWriterSettings settings)
+        {
+            if (string.IsNullOrEmpty(settings.DestinationPath))
+            {
+                throw new ArgumentNullException("destinationpath");
+            }
+
+            _modulePagePath = settings.DestinationPath;
+            _encoding = settings.Encoding;
+            sb = Constants.StringBuilderPool.Get();
```

Comment:
You get the `StringBuilder` in constructor, so it should ideally be returned in a dispose method.

Reply:
Added Disposing patterning